### PR TITLE
In main code pass various ExecutionContexts instead of using global. WA-73

### PR DESCRIPTION
**Pull request by kshakir**
Use a separate ExecutionContext for business (IO heavy) operations.
Akka routes still use the default dispatcher threads.
Run Health Monitor and Admin Sweeper on their own low throughput threads.
Synchronize Admins and run MongoDB status on ExecutionContext, not current actor event / thread.
Formatting and removing IntelliJ warnings.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by kshakir** [comment]
Tested branch in perf. While mcnulty was running, ran Apache Bench manually on `/status`.

### Loaded

```console
$ SBT_OPTS="-Dconcurrency=50 -DnumSeconds=300 -Denv=perf" sbt compile "gatling:testOnly *AgoraGetAllSimulation"
...
================================================================================
---- Global Information --------------------------------------------------------
> request count                                        100 (OK=100    KO=0     )
> min response time                                   3173 (OK=3173   KO=-     )
> max response time                                  11806 (OK=11806  KO=-     )
> mean response time                                  5170 (OK=5170   KO=-     )
> std deviation                                       1465 (OK=1465   KO=-     )
> response time 50th percentile                       5498 (OK=5498   KO=-     )
> response time 75th percentile                       6062 (OK=6062   KO=-     )
> response time 95th percentile                       6989 (OK=6989   KO=-     )
> response time 99th percentile                       8875 (OK=8875   KO=-     )
> mean requests/sec                                  0.329 (OK=0.329  KO=-     )
---- Response Time Distribution ------------------------------------------------
> t < 800 ms                                             0 (  0%)
> 800 ms < t < 1200 ms                                   0 (  0%)
> t > 1200 ms                                          100 (100%)
> failed                                                 0 (  0%)
================================================================================
```
 
```console
$ ab -n 100 https://agora.dsde-perf.broadinstitute.org/status
This is ApacheBench, Version 2.3 <$Revision: 1826891 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking agora.dsde-perf.broadinstitute.org (be patient).....done


Server Software:        akka-http/10.1.0
Server Hostname:        agora.dsde-perf.broadinstitute.org
Server Port:            443
SSL/TLS Protocol:       TLSv1.2,ECDHE-RSA-CHACHA20-POLY1305,2048,256
TLS Server Name:        agora.dsde-perf.broadinstitute.org

Document Path:          /status
Document Length:        66 bytes

Concurrency Level:      1
Time taken for tests:   16.214 seconds
Complete requests:      100
Failed requests:        0
Total transferred:      146500 bytes
HTML transferred:       6600 bytes
Requests per second:    6.17 [#/sec] (mean)
Time per request:       162.143 [ms] (mean)
Time per request:       162.143 [ms] (mean, across all concurrent requests)
Transfer rate:          8.82 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:       59   94  83.1     67     510
Processing:    49   68  57.2     56     547
Waiting:       48   68  57.2     56     547
Total:        111  162 106.0    124     725

Percentage of the requests served within a certain time (ms)
  50%    124
  66%    130
  75%    137
  80%    145
  90%    265
  95%    473
  98%    562
  99%    725
 100%    725 (longest request)
```

### Overloaded

```console
$ SBT_OPTS="-DrunsPerSec=10 -Dconcurrency=50 -DnumSeconds=30 -Denv=perf" sbt compile "gatling:testOnly *AgoraGetAllSimulation"
...
================================================================================
---- Global Information --------------------------------------------------------
> request count                                        600 (OK=103    KO=497   )
> min response time                                     46 (OK=11547  KO=46    )
> max response time                                  60006 (OK=59943  KO=60006 )
> mean response time                                 46011 (OK=36753  KO=47930 )
> std deviation                                      22811 (OK=13540  KO=23849 )
> response time 50th percentile                      60002 (OK=36971  KO=60003 )
> response time 75th percentile                      60004 (OK=48454  KO=60004 )
> response time 95th percentile                      60005 (OK=57758  KO=60005 )
> response time 99th percentile                      60005 (OK=59815  KO=60005 )
> mean requests/sec                                   4.11 (OK=0.705  KO=3.404 )
---- Response Time Distribution ------------------------------------------------
> t < 800 ms                                             0 (  0%)
> 800 ms < t < 1200 ms                                   0 (  0%)
> t > 1200 ms                                          103 ( 17%)
> failed                                               497 ( 83%)
---- Errors --------------------------------------------------------------------
> i.g.h.c.i.RequestTimeoutException: Request timeout to agora.ds    394 (79.28%)
de-perf.broadinstitute.org/107.178.242.107:443 after 60000 ms
> status.find.is(200), but actually found 502                       102 (20.52%)
> j.n.s.SSLException: handshake timed out                             1 ( 0.20%)
================================================================================
```

```console
$ ab -n 100 https://agora.dsde-perf.broadinstitute.org/status
This is ApacheBench, Version 2.3 <$Revision: 1826891 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking agora.dsde-perf.broadinstitute.org (be patient).....done


Server Software:        
Server Hostname:        agora.dsde-perf.broadinstitute.org
Server Port:            443
SSL/TLS Protocol:       TLSv1.2,ECDHE-RSA-CHACHA20-POLY1305,2048,256
TLS Server Name:        agora.dsde-perf.broadinstitute.org

Document Path:          /status
Document Length:        332 bytes

Concurrency Level:      1
Time taken for tests:   27.231 seconds
Complete requests:      100
Failed requests:        0
Non-2xx responses:      100
Total transferred:      50400 bytes
HTML transferred:       33200 bytes
Requests per second:    3.67 [#/sec] (mean)
Time per request:       272.310 [ms] (mean)
Time per request:       272.310 [ms] (mean, across all concurrent requests)
Transfer rate:          1.81 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:       58  105 196.4     65    1381
Processing:    42  167 912.9     46    9033
Waiting:       42  167 912.9     46    9033
Total:        101  272 935.1    112    9098

Percentage of the requests served within a certain time (ms)
  50%    112
  66%    115
  75%    118
  80%    120
  90%    135
  95%   1138
  98%   1719
  99%   9098
 100%   9098 (longest request)
```

File: src/main/scala/org/broadinstitute/dsde/agora/server/ServerInitializer.scala [hunk 1]
```diff
@@ -12,30 +13,43 @@ import org.broadinstitute.dsde.agora.server.webservice.ApiService
...
 
 
   // create these outside of the startAllServices def, so we have global access to their vals
   private val dbStatus = new AgoraDBStatus(permsDataSource)
-  val healthMonitor = actorSystem.actorOf(HealthMonitor.props(Set(Database, Mongo)) { () =>
-    Map(Database -> dbStatus.mysqlStatus, Mongo -> dbStatus.mongoStatus)
-  }, "health-monitor")
+  val healthMonitor: ActorRef = actorSystem.actorOf(HealthMonitor.props(Set(Database, Mongo))({ () =>
+    Map(Database -> dbStatus.mysqlStatus(), Mongo -> dbStatus.mongoStatus())
+  }).withDispatcher("dispatchers.health-monitor-dispatcher"), "health-monitor")
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/agora/server/ServerInitializer.scala]: use variable `healthMonitorDispatcherName` instead of string name?

**Review by davidangb — APPROVED** [review]
:+1:  lgtm, with one minor comment  inline.

Big issue for me, since I was burned by it:  if we release this to prod and everything goes sour, what is the plan to revert?

**Review by aednichols — COMMENTED** [review]
Potentially annoying but would ease rollback:

Adjust code so that if the new config keys are missing, it defaults to the previous model.

This would make a rollback trivial (edit configs on instances) compared to editing configs AND installing a new (well, old) Docker image.

**Review by aednichols — APPROVED** [review]

**Comment by kshakir** [comment]
One should NOT have to change the config to roll back or forth between the previous version and this one. A revert should Just Work™. The code changes contain sensible defaults within the reference.conf, and previous versions don't use these new stanzas at all.

**Review by aednichols — APPROVED** [review]

**Pull request merged**